src/compiler/emitter.ts

```diff
@@ -1639,10 +1639,12 @@ module ts {
             }
 
             if (root) {
+                currentSourceFile = root;
                 emit(root);

```

Comment:
It used to. This change was @CyrusNajmabadi's idea, but the purpose is to set it earlier, so that it's easy to see it. In other words, there is no reason to wait until emit to set currentSourceFile. If you feel strongly, I can change it back.